pkg/health/health_hpa.go

```diff
@@ -0,0 +1,69 @@
+package health
+
+import (
+	"encoding/json"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type hpaConditions struct {
+	Type    string
+	Reason  string
+	Message string
+}
+
+func is_degraded(condition *hpaConditions) bool {
+	degraded_states := []hpaConditions{
+		{Type: "AbleToScale", Reason: "FailedGetScale"},
+		{Type: "AbleToScale", Reason: "FailedUpdateScale"},
+		{Type: "ScalingActive", Reason: "FailedGetResourceMetric"},
+		{Type: "ScalingActive", Reason: "InvalidSelector"},
+	}
+	for _, degraded_state := range degraded_states {
+		if condition.Type == degraded_state.Type && condition.Reason == degraded_state.Reason {
+			return true
+		}
+	}
+	return false
+}
+
+func getAutoscalingv1HPAHealth(obj *unstructured.Unstructured) (*HealthStatus, error) {
+	healthyStatus := &HealthStatus{
+		Status:  HealthStatusHealthy,
+		Message: "",
+	}
+
+	annotation, ok := obj.GetAnnotations()["autoscaling.alpha.kubernetes.io/conditions"]
```

Comment:
As https://github.com/argoproj/argo-cd/issues/4995 mentions, HPA might have conditions stored under `status`. We need to check both annotation and conditions field in the status.